### DISCUSSION TITLE
Change Font and Colour of Login Box?

### DISCUSSION THREAD

**Discussion by swarfendor437010** [realmazharhussain/gdm-settings#152] [Ideas]
Is it possible to change the format of the user Login Box to have different font/font colour, not just the top panel, but the actual login credentials box and also colour of box?

**Comment by realmazharhussain** [comment]
Currently, no. At least not as an option in gdm-settings.

However, you can apply a modified the shell theme to achieve this.

**Reply by ymilly** [reply]
Thanks for letting us know. Is this a limitation of GDM or is it something that can be built into your gdm-settings tool?

**Reply by realmazharhussain** [reply]
This can be built into GDM Settings but it would be a maintenance burden.

Theme and styling of GDM is internal to GNOME Shell and expected never to be touched by something like this app. So, GNOME Shell developers change names/classes/ids of CSS elements quite often (probably every release). For each theme modification option in GDM Settings, it has to keep up with the changes in GNOME Shell for each release of GNOME Shell.

**Reply by ymilly** [reply]
Thanks for explaining. I looked at the Shell options in gdm-settings but only see Default and default-pure. Would be able to point me in the direction of how I can "apply a modified the shell theme to achieve this." :sweat_smile:

**Reply by realmazharhussain** [reply]
You can create a file `/usr/local/share/themes/custom/gnome-shell/gnome-shell.css`. You can put some css in that file to customize any part of GNOME Shell including the login box. Once that file is created, you will see an option for a shell theme named "custom" in GDM Settings. Apply that and your changes should take effect.

As a reference for your custom changes, you can look at the default css in `/usr/local/share/themes/default-pure/gnome-shell/gnome-shell.css` or `/usr/local/share/themes/default-pure/gnome-shell/gnome-shell-dark.css`; whichever one exists. As an alternative, you can also look at [the source code of the shell theme](https://gitlab.gnome.org/GNOME/gnome-shell/-/tree/main/data/theme) right from its GitLab repo but be aware that the shell repo contains SASS but you need to write CSS.
